Library/Services/ContextStorage.php

```diff
@@ -227,10 +241,52 @@ public function getWorkflowStepList(string $workflowProcessId) : array {
      * @returns array
      */
     public function setWorkflowStepList(string $workflowProcessId, $workflowStepList) {
+        $this->di->get('logger')->debug(json_encode(func_get_args()));
         $contextHash = $this->get($workflowProcessId);
         $contextHash[self::WORKFLOW][self::WORKFLOW_STEPS] = $workflowStepList;
         $this->save($workflowProcessId, $contextHash);
     }
 
+    /**
+     * Update the given step related to the given workflow/stepcode
+     *
+     * @param string $workflowProcessId the workflow identifier
+     * @param string $stepCode          the step code
+     * @param array  $stepHash          the step data to save
+     *
+     * @returns array
+     */
+    public function updateWorkflowStep(string $workflowProcessId, $stepCode, $stepHash)
+    {
+        $this->di->get('logger')->debug(json_encode(func_get_args()));
+        $script = <<<EOT
+        def nbi=ctx._source.workflow.steps.size();
+        for (i=0; i < nbi; i++) {
+            if (ctx._source.workflow.steps[i] instanceof List) {
+                def nbj=ctx._source.workflow.steps[i].size();
+                for (j=0; j < nbj; j++) {
+                    if (ctx._source.workflow.steps[i][j].name == stepCode) {
+                        ctx._source.workflow.steps[i][j] = stepHash;
+                        break;
+                    }
+                }
+            } else if (ctx._source.workflow.steps[i].name == stepCode) {
+                ctx._source.workflow.steps[i] =stepHash;
+            }
+        }
+EOT;
+        $updateHash = [
```

Comment:
we can add updatePartial  on elasticsearchadapter.

updatePartial(string $workflowProcessId, string updatePartialName, array $paramHash)

switch (updatePartialName)
  case 'updateWorkflowStep':
            $script = <<<EOT....
this->adapter->update($workflowProcessId, $updateHash);

Reply:
not proud too about my first implementation ... 
We obviously have to keep the `ContextStorage` not related to any storage engine.
I did not found a great solution yet ...